app/policies/provider_policy.rb

```diff
@@ -19,7 +19,14 @@ def initialize(user, provider)
     @provider = provider
   end
 
-  def index?
+  def can_list_providers?
+    user.present?
+  end
+
+  def can_manage_provider?
     user.providers.include?(provider)
   end
+
+  alias_method :index?, :can_list_providers?
+  alias_method :show?, :can_manage_provider?
```

Comment:
My first intuition was to do this like so:

```
def index?
  user.present?
end

def show?
   user.providers.include?(provider)
end

alias_method :can_list_courses?, :show?
```

Reply:
Fixed.